# Tip Calculator

This project enables you to calculate tip on a restaurant bill based on the 'x%' of the bill amount which we will taking from the user. This program will also help you split the bill amount based on the number of people you had lunch/dinner with which will also include the tip amount

We'll be looking into different datatypes and how can we convert one datatype to another for calculation purposes. Other concept discussed here is about `f string` which can help us print a string and a number within the same `print` function and can also allow us to use a `variable` directly to achieve the same.

In [1]:
# giving out a welcome message for the program
print("Welcome to tip calculator")

Welcome to tip calculator


In [5]:
bill = float(input("What was the bill amount? (in $)\n"))

What was the bill amount? (in $)
150


- Taking the bill amount as an input, while doing that we are also converting the `text` input taken from user to a `float` datatype and storing it in a variable

In [15]:
tip_actual = int(input("What % of the bill would you like to tip? 10, 12 or 15? (in %)\n")) 
tip_converted = (tip_actual / 100) + 1

What % of the bill would you like to tip? 10, 12 or 15? (in %)
10


- Here we are doing two things
    
    1. Taking the text input from the user his/her desired % of bill amount to be used for tip
    
    2. We are converting the % of tip into a multiplying factor which we will be directly using to multiply with the split amount that we get for each individual
    For ex. Let's assume the bill amount is $150 and 12% of the amount is supposed to be the tip amount. 
    
    Hence,12% of 150 = (12 / 100) x 150 = 0.12 * 150 = $18
    
    
- Assuming that the bill is getting divided into 4 people, the amount that needs to be divided into 4 people would be 150 + 18 = $168
    
  
- Shorter way of doing this would be multiplying 150 with 1.12, where 1 represent the 150 and 0.12 on top of that. Hence, in the `tip` variable above, we divide the input number by 100 and then add 1 to make it 1.12

In [10]:
# taking input for number of people the bill should be split into
people = int(input("How many people are splitting the bill?\n"))

How many people are splitting the bill?
4


In [23]:
# split_amount = round(((bill / people) * tip),2)
split_amount = "{:.2f}".format((bill / people) * tip)

- the `{:.2f}` tells the code to take the number upto 2 decimal places

In [24]:
print(f'''
Your bill amount is ${bill}
Your bill amount after including {tip_actual}% tip will be ${bill*tip}
After splitting the bill into {people} people, each person would have to pay: ${split_amount}

''')


Your bill amount is $150.0
Your bill amount after including 10% tip will be $165.0
After splitting the bill into 4 people, each person would have to pay: $41.25


